### PR TITLE
Add script for generating docs

### PR DESCRIPTION
**Pull request by ry**
typedoc is very bad - but it's better than nothing.
Website is at http://deno.land/

### CODE REVIEW

**Comment by kitsonk** [comment]
LGTM and something is better than nothing.

As a side note, I have a branch where I was going to emit a fairly straightforward JSON structure that we could use to generate static content: https://github.com/kitsonk/deno/tree/docs using the same infrastructure we use to generate the runtime type library.

**Comment by kitsonk** [comment]
Refs #573

File: website/app.js [hunk 1]
```diff
@@ -182,6 +182,13 @@ function formatSecsAsMins(t) {
  * @param dataUrl The url of benchramk data json.
  */
 export function drawCharts(dataUrl) {
+  // TODO Using window["location"]["hostname"] instead of
+  // window.location.hostname because when deno runs app_test.js it gets a type
```
Comment (piscisaureus) [inline website/app.js]: Or just use "declare var" ?

File: tools/docs.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+# Builds into target/doc
+run(["cargo", "doc", "--no-deps", "-vv"])
+
+# 'deno --types' is stored in target/debug/gen/lib/lib.deno_runtime.d.ts
+# We want to run typedoc on that declaration file only.
+os.chdir(os.path.join(target_path, "debug/gen/lib/"))
+
+# You must have typedoc installed seprately.
+# TODO Replace typedoc with something else ASAP. It's very awful.
```
Comment (piscisaureus) [inline tools/docs.py]: Propel's gendoc not working?

**Pull request merged**